Mathlib/Data/Matrix/Composition.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2024 Yunzhou Xie. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Yunzhou Xie
+-/
+
+import Mathlib.Data.Matrix.Basic
+
+/-!
+## Main results
+# Composition of matrices
+This file shows that Mₙ(Mₘ(R)) ≃ Mₙₘ(R), Mₙ(Mₘ(R)) ≃ Mₘ(Mₙ(R)), Mn(Rᵒᵖ) ≃ₐ[K] Mₙ(R)ᵒᵖ
+and also different levels of equivalence when R is an AddCommMonoid,
+Semiring, and Algebra over a CommSemiring K.
+-/
+
+namespace Matrix
+
+variable  (I J K L R : Type*)
+
+/-- Mₙ(Mₘ(R)) ≃ Mₙₘ(R) -/
+@[simps]
+def comp : Matrix I J (Matrix K L R) ≃ Matrix (I × K) (J × L) R where
+  toFun m ik jl := m ik.1 jl.1 ik.2 jl.2
+  invFun n i j k l := n (i, k) (j, l)
+  left_inv _ := rfl
+  right_inv _ := rfl
+
+/-- Mₙ(Mₘ(R)) ≃ Mₘ(Mₙ(R)) -/
```

Comment:
This docstring does not accurately reflect the declaration at all; you are not composing the `Matrix` function with itself here. It's not clear to me that this declaration should exist at all; it's just `Matrix.reindex (.prodComm _ _) (.prodComm _ _)`. So I'm not entirely clear on whether making a new definition (which needs unfolding and API lemmas etc etc if it's made) is the best approach here. Even if it does then it's not clear to me that it should be in this file because it's not about composing `Matrix` with itself.

Reply:
I think u r right we do have reindexAlgEquiv and reindexLinearEquiv

Reply:
I'm just gonna delete them then

Reply:
Agreed, I think this should be deleted.

Reply:
done :)

Reply:
Don't forget to push!

Reply:
pushed :)